cdap-ui/app/features/dashboard/controllers/widget.js

```diff
@@ -73,6 +77,33 @@ angular.module(PKG.name+'.feature.dashboard')
       this.dataSrc.stopPoll(id);
     };
 
+    var roundUpToNearest = function(val, nearest) {
+      return Math.ceil(val / nearest) * nearest;
+    };
+    var roundDownToNearest = function(val, nearest) {
+      return Math.floor(val / nearest) * nearest;
+    };
+
+    var zeroFill = function(resolution, result) {
+        // interpolating (filling with zeros) the data since backend returns only metrics at specific time periods
+        // instead of for the whole range. We have to interpolate the rest with 0s to draw the graph.
+        if (resolution == '1h') {
+          skipAmt = 60 * 60;
+        } else if (resolution == '1m') {
+          skipAmt = 60;
+        } else {
+          skipAmt = 1;
+        }
+
+        var startTime = roundUpToNearest(result.startTime, skipAmt);
```

Comment:
right. you can still round an epoch time to the nearest 60 or the nearest 3600 though.